src/mod/endpoints/mod_sofia/mod_sofia.c

```diff
@@ -586,8 +586,10 @@ switch_status_t sofia_on_hangup(switch_core_session_t *session)
 
 				if (tech_pvt->respond_phrase) {
 					phrase = su_strdup(nua_handle_home(tech_pvt->nh), tech_pvt->respond_phrase);
-				} else {
+				} else if (sip_cause >= 400 && sip_cause < 700) {
 					phrase = sip_status_phrase(sip_cause);
```

Comment:
this will leak!  There is another strdup below!
if (sip_cause >= 500) {
					peer_sip_reason_phrase = switch_channel_get_variable(channel, "peer_sip_reason_phrase");
					if (!zstr(peer_sip_reason_phrase)) {
						strcpy(suffixed_reason_phrase, phrase);
						strcat(suffixed_reason_phrase, " TV1");
						phrase = su_strdup(nua_handle_home(tech_pvt->nh), suffixed_reason_phrase);
					}
				}

Reply:
This is allocated on nua mem pool. We need to allocate this when we to append tv1 due to we are sending this variable to nua_respond in which will be pushed to nua thread thus we need this allocated in pool. Otherwise the phrase will be a garbage value once its processed in nua thread.